### PR TITLE
Utilize pagination when querying for repos in an org

### PR DESCRIPTION
**Pull request by mprahl**
The default per_page was 30 and when there are more than 30 repos, there are repos that are missing from the returned query.

### CODE REVIEW

**Review by dhaiducek — APPROVED** [review]
Kind of weird there's no pagination metadata to let you know whether to send another query, but I guess it's simple enough to figure out...

File: src/syncer.ts [hunk 1]
```diff
@@ -279,31 +279,63 @@ export class Syncer {
...
-            if (err.status === 404) {
-              this.logger.debug(`Could not find the org ${org}, assuming it's a user`);
-              return;
-            }
+        let isUserRepo = false;
+        let page = 1;
 
-            throw err;
-          });
+        while (true) {
```
[the remark this answers is not in the document]
Comment (mprahl) [reply src/syncer.ts:292]: It's a good suggestion but then I'd have to declare `resp` outside of the loop for the `while` part to have access to it.

File: src/syncer.ts [hunk 2]
```diff
@@ -279,31 +279,63 @@ export class Syncer {
...
+          repos = repos.concat(resp.data);
+
+          page++;
+        }
+
+        if (isUserRepo) {
+          page = 1;
+
+          while (true) {
+            const resp = await this.orgs[installationOrg].client?.repos.listForUser({
```
Comment (yiraeChristineKim) [inline src/syncer.ts:322]: is there any error catch? such as try catch block or .catch(handleError);

Comment (yiraeChristineKim) [reply src/syncer.ts:322]: for all await

Comment (mprahl) [reply src/syncer.ts:322]: This is run as part of a one time initialization at start up and if this fails, there is no way to recover which is why there is nothing catching the error.

**Pull request merged**